api/novelist.py

```diff
@@ -433,6 +459,58 @@ def get_recommendations(self, metadata, recommendations_info):
                 metadata.recommendations += self._extract_isbns(book_info)
         return metadata
 
+    def put_isbns_novelist(self, library):
+        collectionList = []
+        for c in library.collections:
+            collectionList.append(c.id)
+
+        i1 = aliased(Identifier)
+        i2 = aliased(Identifier)
+        e = aliased(Equivalency)
+        lp = aliased(LicensePool)
+
+        isbnQuery = select(
+            [i2.identifier]
+        ).select_from(
+            join(lp, i1, i1.id==lp.id)
+            .join(e, i1.id==e.input_id)
+            .join(i2, e.output_id==i2.id)
+        ).where(
+            and_(lp.collection_id.in_(collectionList), i2.type=="ISBN")
+        ).alias('lp_isbns')
+
+        result = self._db.execute(isbnQuery)
+        isbns = []
+        for r in result:
+            isbns.append(r[0])
+
+        if isbns:
+            response = self.put(
+                self.COLLECTION_DATA_API,
+                { "AuthorizedIdentifier": self.authorized_identifier },
+                data=self.make_novelist_data_object(isbns)
+            )
```

Comment:
It'll be easier to test this if the code to pull the ISBNs from the database is a different method than the code to make the PUT request.